src/Compilers/CSharp/Portable/Syntax/GenericNameSyntax.cs

```diff
@@ -15,5 +16,12 @@ public bool IsUnboundGenericName
                 return this.TypeArgumentList.Arguments.Any(SyntaxKind.OmittedTypeArgument);
             }
         }
+
+        internal override string ErrorDisplayName()
+        {
+            var pb = PooledStringBuilder.GetInstance();
+            pb.Builder.Append(Identifier.ValueText).Append("<").Append(new string(',', Arity - 1)).Append(">");
```

Comment:
Why is this the error display name?  It seems appropriate for an unbound generic. However, if this is a bound generic, why not have the type argument list?